### PR TITLE
Add to package score for having recent maintenance activity

### LINKED ISSUE #1393: Incorporate `lastActivityAt` into the package scoring calculation
We should use this new field as an indicator:

* Last active within 1 month: 15 points
* Last active within 6 months: 10 points
* Last active within a year: 5 points
* Otherwise no extra points

### PR DESCRIPTION
**Pull request by daveverwer**
Fixes #1393

### CODE REVIEW

**Comment by daveverwer** [comment]
Converting to draft. This doesn't work 😅

File: Sources/App/Core/Score.swift [hunk 1]
```diff
@@ -61,7 +64,20 @@ enum Score {
             case .some(3..<5): score += 2
             default: break
         }
-        
+
+        // Last maintenance activity
+        if let lastActivityAt = candidate.lastActivityAt {
+            // Note: This is not the most accurate method to calculate the number of days between
+            // two dates, but is more than good enough for the purposes of this calculation.
+            let dateDifference = Calendar.current.dateComponents([.day], from: lastActivityAt, to: Date())
```
Comment (finestructure) [inline Sources/App/Core/Score.swift:72]: You'd want `Current.date()` here instead of `Date()` :)

File: Tests/AppTests/ScoreTests.swift [hunk 2]
```diff
@@ -90,6 +90,38 @@ class ScoreTests: AppTestCase {
                                            isArchived: false,
                                            numberOfDependencies: 2)),
                        82)
+        XCTAssertEqual(Score.compute(.init(supportsLatestSwiftVersion: true,
+                                           licenseKind: .compatibleWithAppStore,
+                                           releaseCount: 20,
+                                           likeCount: 20_000,
+                                           isArchived: false,
+                                           numberOfDependencies: 2,
+                                           lastActivityAt: Calendar.current.date(byAdding: .day, value: -400, to: Date()))),
```
Comment (finestructure) [inline Tests/AppTests/ScoreTests.swift:99]: dto (and below)

**Comment by daveverwer** [comment]
This is tested and working well on staging.

**Review by finestructure — APPROVED** [review]

**Pull request merged**